lang/py/src/avro/schema.py

```diff
@@ -420,15 +419,24 @@ def __eq__(self, that):
     return self.props == that.props
 
 #
-# Decimal Type
+# Decimal Bytes Type
 #
-class DecimalSchema(LogicalSchema):
-  def __init__(self, decimal_type, precision, scale):
-    LogicalSchema.__init__(self, decimal_type, 'decimal')
+
+class BytesDecimalSchema(PrimitiveSchema, LogicalSchema):
+  def __init__(self, schema_type, precision, scale, other_props=None):
+    if schema_type != 'bytes':
+      raise SchemaParseException('Logical type DECIMAL must be backed by fixed or bytes.')
+
+    self._validate({'max_precision': self._max_precision(),
+                   'precision': precision,
+                   'scale': scale})
+
+    PrimitiveSchema.__init__(self, schema_type, other_props)
+    LogicalSchema.__init__(self, 'decimal')
     self.set_prop('precision', precision)
     self.set_prop('scale', scale)
 
-  def validate(self, validation_props=None):
+  def _validate(self, validation_props=None):
     if not isinstance(validation_props.get('precision'), int):
       raise SchemaParseException('Precision (Integer) is required for logical type DECIMAL.')
```

Comment:
nitpick: the exception could indicate "Precision must be an integer." If we check the positive value at the same time, the exception could be "Precision must be a positive integer."